x-pack/plugins/security/public/management/role_mappings/edit_role_mapping/edit_role_mapping_page.tsx

```diff
@@ -211,6 +214,49 @@ export class EditRoleMappingPage extends Component<Props, State> {
     );
   };
 
+  private checkEmptyAnyAllMappings = (obj: RoleMappingRule) => {
+    const arrToCheck: RoleMappingRule[] = [obj];
+
+    while (arrToCheck.length > 0) {
+      const currentObj = arrToCheck.pop();
+      if (typeof currentObj === 'object' && currentObj !== null) {
+        for (const key in currentObj) {
+          if (currentObj.hasOwnProperty(key)) {
+            const value = currentObj[key];
```

Comment:
I have a type error here. I'm not sure what's the best way to handle it
<img width="1352" alt="image" src="https://github.com/user-attachments/assets/fed7d9e4-3a32-4f9a-b92b-960fe73ab4e0">

Reply:
Attempted a fix in https://github.com/elastic/kibana/pull/189340/commits/465c20e9aca725488274cb57bd78c97f58a475e7

Another option - typeguards:

```
const isRoleMappingAnyRule = (obj:any): obj is RoleMappingAnyRule => {
    return 'any' in obj && Array.isArray(obj.any);
}
// add for the other Role mapping rule types
```

Maybe this approach is overkill? Would love some suggestions for this